public/app/core/components/AppChrome/ExtensionSidebar/ExtensionSidebarProvider.tsx

```diff
@@ -0,0 +1,68 @@
+import { createContext, ReactNode, useContext, useEffect, useState, useMemo } from 'react';
+
+import { ComponentTypeWithExtensionMeta, store } from '@grafana/data';
+import { usePluginComponents } from '@grafana/runtime';
+
+export const EXTENSION_SIDEBAR_EXTENSION_POINT_ID = 'grafana/extension-sidebar/v0-alpha';
+const EXTENSION_SIDEBAR_DOCKED_LOCAL_STORAGE_KEY = 'grafana.navigation.extensionSidebarDocked';
+const ENABLED_EXTENSION_SIDEBAR_PLUGINS = ['grafana-investigations-app'];
+
+type ExtensionSidebarContextType = {
+  dockedPluginId: string | undefined;
```

Comment:
Would be nice to have a comment here as it is not right away clear that this controls what plugin is visible/open.